src/main/java/frc/robot/subsystems/scoring/ScoringSubsystem.java

```diff
@@ -0,0 +1,126 @@
+package frc.robot.subsystems.scoring;
+
+import edu.wpi.first.wpilibj.Timer;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.subsystems.shooter.ShooterIOInputsAutoLogged;
+
+public class ScoringSubsystem extends SubsystemBase {
+    private final ShooterIO shooterIo;
+    private final ShooterIOInputsAutoLogged shooterInputs = new ShooterIOInputsAutoLogged();
+
+    private final AimerIO aimerIo;
+    private final AimerIOInputsAutoLogged aimerInputs = new AimerIOInputsAutoLogged();
+
+    private final Timer timer = new Timer();
+
+    private enum ScoringState {
+        WAIT,
+        INTAKE,
+        PRIME,
+        SHOOT,
+        ENDGAME
+    }
+
+    private ScoringState state = ScoringState.WAIT;
+
+    public ScoringSubsystem(ShooterIO shooterIo, AimerIO aimerIo) {
+        this.shooterIo = shooterIo;
+        this.aimerIo = aimerIo;
+    }
+
+    public void intake() {
+        if (state == ScoringState.WAIT) {
+            enterIntake();
+        }
+    }
+
+    public void aim() {
+        if (state == ScoringState.WAIT && true) { // TODO: Add banner sensor
+            enterPrime();
+        }
+    }
+
+    public void shoot() {
+        if (state == ScoringState.PRIME) {
+            enterShoot();
+        }
+    }
+
+    public void abort() {
+        if (state == ScoringState.PRIME || state == ScoringState.SHOOT) {
+            enterWait();
+        }
+    }
+
+    public void endgame() {
+        if (state == ScoringState.WAIT) {
+            enterEndgame();
+        }
+    }
```

Comment:
Let's make each of these functions run the logic for the given state, including the automatic transitions to other states

Reply:
Converted to state logic